include/seqan3/core/tuple_utility.hpp

```diff
@@ -6,182 +6,16 @@
-//!\endcond
-constexpr auto tuple_split(tuple_t<ts...> const & t)
-{
-    static_assert(pivot_c <= sizeof...(ts));
-
-    return tuple_t{detail::tuple_split<0>(t, std::make_index_sequence<pivot_c>{}),
-                   detail::tuple_split<pivot_c>(t, std::make_index_sequence<sizeof...(ts) - pivot_c>{})};
-}
-
-//!\copydoc seqan3::tuple_split
-template <size_t pivot_c, template <typename ...> typename tuple_t, typename ...ts>
-//!\cond
-    requires tuple_like<tuple_t<ts...>>
-//!\endcond
-constexpr auto tuple_split(tuple_t<ts...> && t)
-{
-    static_assert(pivot_c <= sizeof...(ts));
-
-    return tuple_t{detail::tuple_split<0>(std::move(t), std::make_index_sequence<pivot_c>{}),
-                   detail::tuple_split<pivot_c>(std::move(t), std::make_index_sequence<sizeof...(ts) - pivot_c>{})};
-}
-
-/*!\brief Splits a tuple like data structure at the first position of the given type.
- * \ingroup core
- *
- * \tparam    pivot_t A template type specifying the split position.
- * \param[in] t       The original tuple to split.
- *
- * \returns A new tuple of tuples with the left side of the split and the right side of the split.
- *
- * \details
- *
- * Splits a tuple into two tuples, while the element at the split position will be contained in the second tuple.
- * Note, that the returned tuples can be empty. For this reason it is not possible to use tuple like objects,
- * that cannot be empty, i.e. std::pair. Using such an object will emit an compiler error.
- *
- * ### example
- *
- * \include test/snippet/core/tuple_utility.cpp
- *
- * ### Complexity
- *
- * Linear in the number of elements.
- *
- * ### Thread safety
- *
- * Concurrent invocations of this functions are thread safe.
- */
-template <typename pivot_t, tuple_like tuple_t>
-constexpr auto tuple_split(tuple_t && t)
-{
-    constexpr size_t pivot_c = list_traits::find<pivot_t, detail::tuple_type_list_t<std::remove_cvref_t<tuple_t>>>;
-
-    static_assert(pivot_c <= std::tuple_size_v<std::remove_cvref_t<tuple_t>>);
-
-    return tuple_split<pivot_c>(std::forward<tuple_t>(t));
-}
-
-/*!\brief Removes the first element of a tuple.
- * \ingroup core
- *
- * \param[in] t  The original tuple.
- *
- * \returns A new tuple without the first element of `t`.
- *
- * \details
- *
- * Note, that the tuple must contain at least one element and must support empty tuple types, i.e. std::pair cannot
- * be used.
- *
- * ### Complexity
- *
- * Linear in the number of elements.
- *
- * ### Thread safety
- *
- * Concurrent invocations of this functions are thread safe.
- */
-template <tuple_like tuple_t>
-constexpr auto tuple_pop_front(tuple_t && t)
-{
-    static_assert(std::tuple_size_v<std::remove_cvref_t<tuple_t>> > 0);
-
-    return std::get<1>(tuple_split<1>(std::forward<tuple_t>(t)));
-}
-//!\}
-
-} // namespace seqan3
+SEQAN3_DEPRECATED_HEADER(
+   "This header is deprecated and will be removed in SeqAn-3.1.0; "
+   "Please #include <seqan3/utility/tuple_utility.hpp> instead.")
```

Comment:
```suggestion
SEQAN3_DEPRECATED_HEADER(
   "This header is deprecated and will be removed in SeqAn-3.1.0; Please #include <seqan3/utility/tuple_utility.hpp> instead.")
```

No line break here :)

Reply:
ok :-(

:p